File: src/main/java/com/ksptool/ql/biz/model/dto/SaveThemeDto.java
package com.ksptool.ql.biz.model.dto;

import lombok.Data;
import jakarta.validation.constraints.NotNull;

import java.util.Map;

@Data
public class SaveThemeDto {

    // 主题ID
    private Long themeId;

    // 主题名称（可选）
    private String themeName;
    
    // 主题描述（可选）
    private String description;
    
    // 主题值对象
    @NotNull
    private Map<String,String> themeValues;
}

File: src/main/resources/views/aether/src/entity/dto/SaveThemeDto.ts
import type ThemeValuesDto from "@/entity/dto/ThemeValuesDto.ts";

export default interface SaveThemeDto {
    /**
     * 主题ID (可选，用于更新)
     */
    themeId?: string;

    /**
     * 主题名称 (可选)
     */
    themeName?: string;

    /**
     * 主题描述 (可选)
     */
    description?: string;

    /**
     * 主题值映射，key为属性名，value为属性值
     */
    themeValues: ThemeValuesDto;
} 